notebooks/Project 1 - Algorithms.ipynb

```diff
@@ -86,68 +86,74 @@
+      "text/plain": [
+       "<matplotlib.figure.Figure at 0x108b80550>"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
    "source": [
     "import time\n",
     "import math\n",
-    "import matplotlib\n",
     "import matplotlib.pyplot as plt\n",
-    "%matplotlib inline\n",
+    "import sys\n",
+    "sys.setrecursionlimit(2000)\n",
+    "% matplotlib inline\n",
     "\n",
     "# comparision of the time required for each alternative to run\n",
     "def iterative_factorial(n):\n",
     "    factorial = 1\n",
-    "    for i in range(n, 1, -1): # this sets range from the factorial (i.e. 5!) down to 1\n",
-    "        factorial *= n        # factorial is the variable the accumulates the product each loop\n",
+    "    for i in range(n, 1, -1):\n",
+    "        factorial *= n\n",
     "        n -= 1\n",
     "    return factorial\n",
     "\n",
+    "\n",
     "def recursive_factorial(n):\n",
-    "    if n<= 1:                 # exit out of the loop once we get down to 1\n",
+    "    if n < 1:\n",
     "        return 1\n",
-    "    return n * recursive_factorial(n-1) # core of the recursion\n",
+    "    return n * recursive_factorial(n - 1)\n",
     "\n",
-    "def factorial_compare(n):\n",
```

Comment:
A few things with this function:
1. Let your variables speak for themselves. If a variable represents the starting timestamp, just name it `starting_timestamp`. This both reduces clutter and avoids excessive commentary.
2. There's quite a bit of redundancy here. In these cases, identify the objects that are shared, and those that are unique. Try to think if you can create some sort of loop to make it so that you only have to write each operation once.